File: lb_modules/python/littlebit.py
import urllib, urllib2

class LittleBit:
    def __init__(self, api_key):
        self.api_key = api_key

    def donate(self, amount):
        url = 'http://localhost/donations'
        data = urllib.urlencode({
            "api_key": self.api_key,
            "amount": amount
        })
        req = urllib2.Request(url, data)
        rsp = urllib2.urlopen(req)
        return rsp.read()

File: lb_modules/php/littlebit.php
<?php

class LittleBit {

    public function donate ($amount) {

        # Our new data
        $data = array(
            'api_key' => $this->api_key,
            'amount' => $amount
        );

        # Create a connection
        $url = 'localhost/donations';
        $ch = curl_init($url);

        # Form data string
        $postString = http_build_query($data, '', '&');

        # Setting our options
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $postString);
        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);

        # Get the response
        $response = curl_exec($ch);
        curl_close($ch);
        echo $response;
    }

    private $api_key = NULL;

    function __construct ($api_key) {
        $this->api_key = $api_key;
    }
}

?>

